### ISSUE TITLE
Password reset Cognito API error response not caught

### ISSUE THREAD

**Issue by oschwartz10612** [aws-amplify/amplify-js#11456]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Angular

### Amplify APIs

Authentication

### Amplify Categories

auth

### Describe the bug

Auth.currentAuthenticatedUser does not currently catch if the password is reset in Cognito. It currently will just return the current user even though the Cognito API call returns 400 with the following message.

`{"__type":"PasswordResetRequiredException","message":"Password reset required for the user"}`

Fixed by #11274

### Expected behavior

We need to catch this message I believe in the same way that the token revoke is handled.

### Reproduction steps

1. Sign into a demo application
2. From Cognito, reset the users password
3. Reload the page

**Comment by cwomack** [comment]
Hello, @oschwartz10612 👋.  Thank you for opening this issue and already submitting PR #11274 tied to it!  I'm able to reproduce the situation you're describing in this issue, but hoping to get a little more context on what the desired/ideal solution would be.  

Is the concern about security of the logged in user should an admin force reset the password on the Cognito console?  Would you be expected that the user be globally logged out with all tokens revoked when this happens?  Any additional clarity would be appreciated

**Comment by oschwartz10612** [comment]
Hi @cwomack!

I think the issue here is that the API requests fail and we dont get the full user from Cognito because the user is no longer valid, so they need to be kicked back to the login screen to change the password and receive new tokens. I my specific app the user's password gets reset after 30 days for security, so in the middle of a session, it could get invalidated and then need to be kicked back to sign in.

I think it generally makes sense to catch this as a error in currentAuthenticatedUser because if I remember correctly in my testing the user returned is basically useless until the password reset is handled correctly. 

I have had to implement a fairly hacky way to catch this which you can see in the below code. Maybe this will help explain what I mean.

```javascript
    // Get the user on creation of this service
    Auth.currentAuthenticatedUser({ bypassCache: true }).then(
      (user: any) => {
        Auth.getPreferredMFA(user, { bypassCache: true }).then((mfa: any) => { // this needs to be removed once the changes are merged into amplify to handle Password reset required for the user
          this.setUser(user);
          console.log('User signed in from currentAuthenticatedUser');
        }).catch((err: any) => {
          console.log(err);
          Auth.signOut();
          this._authState.next(loadedEmptyAuthState);
          console.log('User not signed in because user is null');
        });
      },
      (err: any) => {
        console.log(err);
        this._authState.next(loadedEmptyAuthState);
        console.log('User not signed in because of error');
      }
    )
```

**Comment by oschwartz10612** [comment]
Hi @cwomack , any news on this? Would it be possible to merge sometime soon? Do you need anything more from me?

**Comment by erinleigh90** [comment]
Hi @oschwartz10612, working on testing the PR and getting it approved. Thank you for your contribution! We appreciate you :).

**Comment by cwomack** [comment]
@oschwartz10612, appreciate your patience while we tested this and got the PR approved.  PR #11274 was merged today with Amplify v5.3.2 if you want to upgrade and test/verify the fix on your side.

One thing to note is that if the `userInfo` is in local storage, calling `Auth.currentAuthenticatedUser()` will not call the server or get the error. This means to properly test it, you'll have to delete `userInfo` from local storage, _**then**_ call `Auth.currentAuthenticatedUser()`.
